### DISCUSSION TITLE
Assignment 9 - data training: error when trying to evaluate a model

### DISCUSSION THREAD

**Discussion by andreicorni25** [Mayo-Radiology-Informatics-Lab/MIDeL#13] [Q & A] [answered]
Hello, could someone guide me in the right direction. I don't understand what is wrong with my code.

Thank you!

`
from skimage.io import imread
from monai.transforms import AddChannel
from skimage.io import imread
import numpy as np
import cv2
from torch.utils.data import DataLoader
from torchvision import models

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

img_array = imread(train_imageinfo_list[0][0])
resized_img = cv2.resize(img_array, (224, 224))
img_tensor = torch.from_numpy(resized_img).to(device)

channel_adder = AddChannel()
channel_image = channel_adder(img_tensor)
channel_repeater = RepeatChannel(3)
img = channel_repeater(channel_image)
img = channel_adder(img)
img.shape

mymodel = build_model(arch = "vgg16", pretrained = False).to(device)
mymodel.eval()
output = mymodel(img)
preds = torch.argmax(torch.softmax(outputs, dim=1), dim=1)
`

> RuntimeError: Input type (torch.cuda.ByteTensor) and weight type (torch.cuda.FloatTensor) should be the same

**Comment by BardiaKh** [comment, marked as answer]
This is caused by the input having a different data type as model weights. I would suggest adding this line just before building your model:

```python
img = img.float()
```

This will convert your image tensor into `float` data type, which is the same as your model weight data type.
